include/RED4ext/ResourceLoader.hpp

```diff
@@ -112,7 +130,7 @@ struct ResourceLoader
         SharedPtr<ResourceToken<T>> token;
         func(this, &token, aPath);
 
-        return token;
+        return std::move(token);
```

Comment:
The one before was good, the compiler will do a copy elision / return value optimization. If I remember right by doing a move here will prevent it.